docs/internals/services.rst

```diff
@@ -1,6 +1,112 @@
-.. include:: /_shared/stub-topic.txt
+Understanding EF Services
+=========================
 
-|stub-icon| Understanding EF Services
-=====================================
+.. contents:: `In this article`
 
-.. include:: /_shared/stub-notice.txt
+Entity Framework executes as a collection of services working together. A
+service is a reusable component. In C#, a services is typically an
+implementation of an interface. Services are is available to other services via
+`dependency injection (DI) <https://wikipedia.org/wiki/Dependency_injection>`_.
+This article covers some fundamentals principals for understanding how EF uses
+services and DI.
+
+Categories of Services
+----------------------
+
+Services fall into one or more categories.
+
+Context services
+  Services that related to interaction with a specific instance of
```

Comment:
Did you mean "that relate to the interaction with a specific instance" or "that are related to a specific instance"?